easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1185,15 +1192,15 @@ def handle_external_module_metadata_by_probing_modules(self, dep_name):
             ('%s_HOME' % short_ext_modname_upper, '%s_VERSION' % short_ext_modname_upper),
         ]
 
-        for p, v in allowed_pairs:
-            prefix = self.modules_tool.get_variable_from_modulefile(dep_name, p)
-            version = self.modules_tool.get_variable_from_modulefile(dep_name, v)
+        for prefix, version in allowed_pairs:
+            module_prefix = self.modules_tool.get_variable_from_modulefile(dep_name, prefix)
+            module_version = self.modules_tool.get_variable_from_modulefile(dep_name, version)
 
-            if prefix and version:
+            if module_prefix and module_version:
                 dependency = {
-                    'name': [short_ext_modname],
-                    'version': [version],
-                    'prefix': p
+                    'name': [short_ext_modname_upper],
```

Comment:
`short_ext_modname` (`= dep_name.split('/')[0]`) seemed to be ok as `dependency['name']`, why the change to `short_ext_modname_upper` (which is used in environment variables)?

Reply:
@migueldiascosta, what I wrote is wrong...
Actually, there is no way around it... The name maps the external module name to EasyBuild's internal one. E.g., if the module is called `cudatoolkit`, then the user has to tell in the metadafile that this represents `CUDA`. So, I changed the function behavior in order to only fill in the missing terms...
This way, the user can set in the metadata file
```
[cudatoolkit]
name = CUDA
```
Then, the code will load the default `cudatoolkit` module and attempt to capture the version and prefix.